docs/rules/string-match.md

```diff
@@ -0,0 +1,44 @@
+# disallow strings that match a given regular expression (string-match)
+
+Strings are gold mines of hidden information: I've seen type systems and enumerations implemented in strings. I've seen numbers being used as strings to get around type conversions. And sometimes, you just want to enforce some consistency in your errors. This rule allows you to highlight these issues and decide how you want to act on them.
+
+## Rule Details
+
+This rule disallows strings that match a specified regular expression.
+
+
+## Options
+
+This rule has a string option for the specified regular expression.
+
+For example, to ensure no fully numeric strings appear:
+
+```json
+{
+    "string-match": ["error", "^[0-9\-\.][0-9\.]*$"]
```

Comment:
I don't think the backslashes in this string are doing what you think they're doing. This is identical to:

```json
{
    "string-match": ["error", "^[0-9-.][0-9.]*$"]
}
```

You probably meant to include an actual backslash character in the string, in which case you would want to use:

```json
{
    "string-match": ["error", "^[0-9\\-\\.][0-9\\.]*$"]
}
```